76.md

```diff
@@ -0,0 +1,90 @@
+NIP-76
+======
+
+Relay Read Permissions
+----------------------
+
+`draft` `optional`
+
+Tag names `rp` (read permission) and `prp` (probabilistic read permission) define which keys are authorized to download an event from the relay. 
+
+Events with an `rp` or `prp` require AUTH to be downloaded.
+
+## Read Permission
+
+The `rp` tag takes a pubkey. Multiple `rp` tags represent a logical OR. 
+
+```json
+["rp", "<pubkey1>"]
+["rp", "<pubkey2>"]
+["rp", "<pubkey3>"]
+```
+
+Relays MUST check if the authed user is one of the keys in the `rp` before sending the event to the client.
+
+## Probabilistic Read Permissions
+
+Probabilistic permissions use bloom filters of a set of authorized pubkeys. They are represented by a colon-separated value with: 
+1. the number of bits in the bit array
+2. the number of hashing rounds used by the filter 
+3. the bit array in Base64.
+
+```json
+["prp", "<bits>:<rounds>:<base64>"]
```

Comment:
Apart from obvious issues of a probabilistic access control and access for more than two parties in general, probabilistic filters can easily be brute-forced even if relays honor them.

As the event itself would store the parameters of the Bloom Filter, it could already vary bits and rounds to through off an attacker but it's certainly not meant for that.

I would add a `<salt>` parameter here that gets used in the hashing as `return BigInt(sha256(value || salt || index)) % size`